src/write.c

```diff
@@ -2306,46 +2252,62 @@ static avifBool avifEncoderIsMinimizedImageBoxCompatible(const avifEncoder * enc
     }
 #endif
 
-    // 4:4:4, 4:2:2, 4:2:0 and 4:0:0 are supported by a MinimizedImageBox.
+    if (encoder->data->imageMetadata->width > (1 << 15) || encoder->data->imageMetadata->height > (1 << 15)) {
+        return AVIF_FALSE;
+    }
+    if (encoder->data->imageMetadata->icc.size > (1 << 20) || encoder->data->imageMetadata->exif.size > (1 << 20) ||
+        encoder->data->imageMetadata->xmp.size > (1 << 20)) {
+        return AVIF_FALSE;
+    }
+
+    // 4:4:4, 4:2:2, 4:2:0 and 4:0:0 are supported by a MetaBox with version 1.
     if (encoder->data->imageMetadata->yuvFormat == AVIF_PIXEL_FORMAT_NONE) {
         return AVIF_FALSE;
     }
 
+    if (encoder->data->imageMetadata->colorPrimaries > 255 || encoder->data->imageMetadata->transferCharacteristics > 255 ||
+        encoder->data->imageMetadata->matrixCoefficients > 255) {
+        return AVIF_FALSE;
+    }
+
     const avifEncoderItem * colorItem = NULL;
     for (uint32_t itemIndex = 0; itemIndex < encoder->data->items.count; ++itemIndex) {
         avifEncoderItem * item = &encoder->data->items.item[itemIndex];
 
-        // Grids are not supported by a MinimizedImageBox.
+        // Grids are not supported by a MetaBox with version 1.
         if (item->gridCols || item->gridRows) {
             return AVIF_FALSE;
         }
 
         if (item->id == encoder->data->primaryItemID) {
             assert(!colorItem);
             colorItem = item;
-            continue; // The primary item can be stored in the MinimizedImageBox.
+            if (item->encodeOutput->samples.count != 1 || item->encodeOutput->samples.sample[0].data.size > (1 << 28)) {
+                return AVIF_FALSE;
+            }
+            continue; // The primary item can be stored in the MetaBox with version 1.
         }
         if (item->itemCategory == AVIF_ITEM_ALPHA && item->irefToID == encoder->data->primaryItemID) {
-            continue; // The alpha auxiliary item can be stored in the MinimizedImageBox.
+            if (item->encodeOutput->samples.count != 1 || item->encodeOutput->samples.sample[0].data.size >= (1 << 28)) {
```

Comment:
We use `>` in line 2285 but `>=` here. Should these two conditional expressions be exactly the same? Or is the maximum size of the alpha item smaller?

Reply:
I understand this now. This is because one has `minus_one` and the other does not:

```
    unsigned int(item_data_size_size) main_item_data_size_minus_one;
    if (has_alpha)
        unsigned int(item_data_size_size) alpha_item_data_size;
```